File: lab 7/lab6/FromSocket.java
package lab6;

public class FromSocket extends ElectricalDevice {
    private boolean isPowerOn;
    private int power;

    public FromSocket(String name, int electromagneticRadiation, int power, boolean isPowerOn) {
        super(name, electromagneticRadiation);
        this.power = power;
        this.isPowerOn = isPowerOn;
    }

    public int getPower() {
        return this.power;
    }

    public void setPower(int newPower) {
        this.power = newPower;
    }

    public boolean getIsPowerOn() {
        return this.isPowerOn;
    }
}

File: lab 6/FromSocket.js
const ElectricalDevice = require("./ElectricalDevice");

class FromSocket extends ElectricalDevice {
    #isPowerOn;
    #power;
    constructor(name, electromagneticRadiation, power, isPowerOn) {
        super(name, electromagneticRadiation);
        this.#power = power;
        this.#isPowerOn = isPowerOn || false;
    }

    get power() {
        return this.#power
    }

    set power(newPower) {
        this.#power = newPower;
    }

    showPower() {
        return `${this.name} power=${this.power}`
    }

    get isPowerOn() {
        return this.#isPowerOn;
    }

}

module.exports = FromSocket;